setup/steps/check_platform.go

```diff
@@ -0,0 +1,36 @@
+package steps
+
+import (
+	"errors"
+
+	s "github.com/pganalyze/collector/setup/state"
+	"github.com/shirou/gopsutil/host"
+)
+
+var CheckPlatform = &s.Step{
+	Description: "Check platform",
+	Check: func(state *s.SetupState) (bool, error) {
+		hostInfo, err := host.Info()
+		if err != nil {
+			return false, err
+		}
+		state.OperatingSystem = hostInfo.OS
+		state.Platform = hostInfo.Platform
+		state.PlatformFamily = hostInfo.PlatformFamily
+		state.PlatformVersion = hostInfo.PlatformVersion
+
+		if state.Platform == "ubuntu" {
+			if state.PlatformVersion < "14.04" {
+				return false, errors.New("Ubuntu versions older than 14.04 are not supported")
+			}
+		} else if state.Platform == "debian" {
+			if state.PlatformVersion < "10.0" {
```

Comment:
This would not sort correctly for version `9.0` (since lexically that would come after `10.0`) - as discussed it might work if we parse these as floats and then compare them.